crates/polars-arrow/src/legacy/kernels/time.rs

```diff
@@ -70,3 +70,16 @@ pub fn convert_to_naive_local_opt(
         LocalResult::None => None,
     }
 }
+
+#[cfg(feature = "timezones")]
+pub fn convert_to_new_timezone_and_naive_local(
```

Comment:
not sure about this name either

how about
- first, in a separate PR, you rename `convert_to_naive_local` to `replace_time_zone_in_naive_utc`. We're going from one naive utc representation to another, so maybe that's a more appropriate name (maybe also add a docstring with an example? optional)
- then, here, this one could be called `naive_utc_to_naive_local_in_new_time_zone`?

Reply:
one sec, before you do too much work here, just checking that there's no opposition to this feature

Reply:
Sure :)